californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -204,18 +208,33 @@ public CoapEndpoint(InetSocketAddress address, NetworkConfig config) {
 	 * @param config the config
 	 */
 	public CoapEndpoint(Connector connector, NetworkConfig config) {
-		this(connector,config, null, null);
+		this(connector, config, null);
 	}
 
-	public CoapEndpoint(InetSocketAddress address, NetworkConfig config, NotificationListener notificationListener,
-			ObserveRequestStore store) {
-		this(createUDPConnector(address, config), config, notificationListener, store);
+	/**
+	 * Instantiates a new endpoint with the specified address and configuration
+	 * and observe request store (used to persist observation).
+	 *
+	 * @param connector the connector
+	 * @param config the config
+	 */
+	public CoapEndpoint(InetSocketAddress address, NetworkConfig config, ObserveRequestStore store) {
+		this(createUDPConnector(address, config), config, store);
 	}
 
-	public CoapEndpoint(Connector connector, NetworkConfig config,NotificationListener notificationListener, ObserveRequestStore store) {
+	/**
+	 * Instantiates a new endpoint with the specified connector and configuration
+	 * and observe request store (used to persist observation).
+	 *
+	 * @param connector the connector
+	 * @param config the config
+	 */
+	public CoapEndpoint(Connector connector, NetworkConfig config, ObserveRequestStore store) {
 		this.config = config;
 		this.connector = connector;
-		this.matcher = new Matcher(config, notificationListener, store);
+		if (store == null)
+			store = new InMemoryObserveRequestStore();
```

Comment:
Well, I would. This is a public method and you have no control over of what clients will pass in and do with the object reference after your constructor has completed.
Personally, I would **always** stick to this principle. But since we have no general coding rule in place for this, I guess you are free to do it any way you like and I do not want to start a broad discussion about it. Just my personal opinion ...